### DISCUSSION TITLE
Spigot Support?

### DISCUSSION THREAD

**Discussion by Kamenzee** [GameServerManagers/LinuxGSM#3517] [General]
Hey guys!

I tried posting this earlier so if this is a double post, I apologize in advance.

To elaborate on my question I had been using MineOS to host the MC server, however, it became too much of a hassle.  I noticed it had very poor performance on the standard MC jar profiles so I was using Spigot (after having to manually FTP the files from the PC to the server due to MineOS not downloading the proper ones).  Either way, this boosted the performance DRAMATICALLY.  No issues whatsoever.

Well, I've used LGSM for all of my other servers which would be, 7dtd, Project Zomboid, ARK, etc... and have had WONDERFUL performance.  I was just wondering if it was possible to use the Spigot upgrade with linuxGSM as well?  If so, can someone help me do so?

**Comment by dgibbs64** [comment]
We currently support PaperMC . Have you tried that?

**Reply by Kamenzee** [reply]
I have not!

I have a pretty beefy server so it should be able to run, but I guess a MC server is more demanding than it seems.  

As far as paperMC it looks like it runs like another standard LGSM server and has some benefits!  I'll give it a shot and see what happens.

While I have someone here, if this needs to be another question then I will post it, but do you by chance know how to run a previous version?  I have friends that want to play on VR and it seems that 1.17.1 does not support VR just yet.

**Reply by dgibbs64** [reply]
PaperMC is meant to be a very efficient version of MC: Java Edition. I wrote an article about the maint versions of MC:JE as I was confused at first. https://danielgibbs.co.uk/2021/02/minefield-minecraft-mods-plugins-and-servers-explained/

You can download older versions of the MC server online and replace the older version with your current version. There is no automated way in LinuxGSM currently.

**Reply by Kamenzee** [reply]
I was just FTPing the 1.16.5.jar from papermc.io as I saw this reply!  lol  I don't mind moving things manually.

That article is very well written and explanatory, you did a wonderful job.  It's great to know how Waterfall and BungeeCord work, too!  That was a great addition.  I want to try that stuff out as well to really expand the server with games.  I also noticed that you worked on the installer, and for that, I want to say thank you for your hard work.  I'd love to help myself, but I'm only a Linux novice and know VERY little about coding languages to mod.
